rules/ensure_default_tags_rule.go

```diff
@@ -59,20 +79,120 @@ func (r *EnsureDefaultTagsRule) Check(runner tflint.Runner) error {
+		awsRunner := &AWSRunner{
+			Runner: runner,
+			Issues: []helper.Issue{},
+		}
+
+		// This runs the AWS Ressource Missing Tags Rule with our custom runner (so it uses our configuration and captures issues)
+		// https://github.com/terraform-linters/tflint-ruleset-aws/blob/master/docs/rules/aws_resource_missing_tags.md
+		err := awsRules.NewAwsResourceMissingTagsRule().Check(awsRunner)
+		if err != nil {
+			return err
+		}
+
+		// If there are issues with both missing default tags and missing resource tags, output all issues found
+		if len(providerDefaultTagIssues) > 0 && len(awsRunner.Issues) > 0 {
+			for _, issue := range providerDefaultTagIssues {
+				err := runner.EmitIssue(issue.Rule, issue.Message, issue.Range)
+				if err != nil {
+					return err
+				}
+			}
+
+			for _, issue := range awsRunner.Issues {
+				err := runner.EmitIssue(issue.Rule, issue.Message, issue.Range)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+	}
+
+	return nil
+}
+
+// Takes a Terraform block and verifies that the block has all the required tags
+func (r *EnsureDefaultTagsRule) verifyRequiredTags(block *hclext.Block, runner tflint.Runner) {
+	tagsAttribute := block.Body.Attributes["tags"]
+	if tagsAttribute == nil {
+		return
+	}
+
+	tagsBlock := tagsAttribute.Expr
+	tagAttributes, _ := tagsBlock.Value(&hcl.EvalContext{})
 
-	for _, resource := range resources.Blocks {
-		defaultTagsBlock := resource.Body.Blocks.OfType("default_tags")
+	if !tagAttributes.Type().IsObjectType() {
+		return
+	}
 
-		if len(defaultTagsBlock) == 0 {
-			return runner.EmitIssue(
-				r,
-				"default_tags is missing",
-				resource.DefRange,
-			)
+	var missingTags []string = []string{}
+	attributeMap := tagAttributes.AsValueMap()
+	for _, requiredTag := range config.Tags {
+		found := false
+		for key := range attributeMap {
+			if key == requiredTag {
+				found = true
+				break
+			}
+		}
+		if !found {
+			missingTags = append(missingTags, requiredTag)
 		}
 	}
 
+	if len(missingTags) > 0 {
+		err := runner.EmitIssue(
+			r,
+			fmt.Sprintf("The provider is missing the following tags: %s.", "\""+strings.Join(missingTags, "\", "+"\"")+"\""),
+			tagsBlock.Range(),
+		)
+		if err != nil {
+			return
+		}
+	}
+}
+
+// AWS runner overrides EmitIssue and DecodeRuleConfig to wrap AwsResourceMissingTagsRule. This allows us to capture the issues it finds as well as use our own configuration
+type AWSRunner struct {
+	tflint.Runner
+	Issues []helper.Issue
+}
+
+func (r *AWSRunner) EmitIssue(rule tflint.Rule, message string, issueRange hcl.Range) error {
+	r.Issues = append(r.Issues, utils.NewIssue(rule, message, issueRange))
+	return nil
+}
+
+func (r *AWSRunner) DecodeRuleConfig(ruleName string, ret interface{}) error {
+	v := reflect.ValueOf(ret).Elem()
+	v.Set(reflect.ValueOf(struct {
+		Tags    []string `hclext:"tags"`
+		Exclude []string `hclext:"exclude,optional"`
+	}{
+		Tags:    config.Tags,
+		Exclude: config.Exclude,
+	}))
```

Comment:
Won't this override the existing `ret` (which I assume is the value of the decoded config)?